torch/_functorch/_aot_autograd/autograd_cache.py

```diff
@@ -509,6 +533,8 @@ def wrap_post_compile(
 
         compiled_fw_func = self.compiled_fw.load(args, fx_config)
         compiled_bw_func = None
+        initialize_cuda_on_backward()
```

Comment:
this is only running on cache hit, right?

As written, will this also run even for cache hit models that are under no_grad? Maybe we should move it to right before `self.compiled_bw.load`

Reply:
Yeah, good idea